tests/trainer/test_trainer.py

```diff
@@ -1127,3 +1128,23 @@ def setup(self, stage):
     trainer.test(ckpt_path=None)
     assert trainer.stage == 'test'
     assert trainer.get_model().stage == 'test'
+
+
+@pytest.mark.parametrize("trainer_kwargs,expected", [
+    pytest.param(dict(), PassThroughProfiler),
+    pytest.param(dict(profiler=False), PassThroughProfiler),
+    pytest.param(dict(profiler=True), SimpleProfiler),
+    pytest.param(dict(profiler=SimpleProfiler()), SimpleProfiler),
+    pytest.param(dict(profiler=AdvancedProfiler()), AdvancedProfiler),
+    pytest.param(dict(profiler="simple"), SimpleProfiler),
+    pytest.param(dict(profiler="advanced"), AdvancedProfiler),
+])
+def test_trainer_profiler_correct_args(trainer_kwargs, expected):
+    trainer = Trainer(**trainer_kwargs)
```

Comment:
I think this is cleaner :smile: 

```suggestion
@pytest.mark.parametrize(['input', 'expected'], [
    (None, PassThroughProfiler),
    (False, PassThroughProfiler),
    (True, SimpleProfiler),
    (SimpleProfiler(), SimpleProfiler),
    (AdvancedProfiler(), AdvancedProfiler),
    ("simple", SimpleProfiler),
    ("advanced", AdvancedProfiler),
])
def test_trainer_profiler_correct_args(input, expected):
    kwargs = {'profiler': input} if input else {}
    trainer = Trainer(**kwargs)
```

Reply:
yep, much cleaner thanks :)